replica/database.hh

```diff
@@ -318,6 +318,7 @@ struct table_stats;
 using column_family_stats = table_stats;
 
 class database_sstable_write_monitor;
+class compaction_group;
```

Comment:
Shouldn't it be in the sstables namespace?
(Maybe we need a compaction namespace by the way, but that's orthogonal to this change)

Reply:
Actually, I see that `table_state` is already defined in namespace compaction.
It makes sense to do the same for `compaction_group`.

Reply:
The idea is that table_state becomes compaction_group_state. compaction_group is a replica entity, rather than one which lives in compaction namespace. We did tons of effort in compaction layer to not reference table directly, but rather work with the table_state. compaction group stores memtable_list and sstable_set's, all of which belong to the replica namespace.

I was thinking yesterday about your question, and perhaps I should create replica/compaction_group.hh and replica/compaction_group.cc, which allows for better organization of table and compaction group code.

Sounds good to you?

Reply:
The organization makes sense.
Unfortunately the name doesn't.
I think this entity is more than about compaction so it's name is misleading.
For example this group of tokens will be a unit for replication purposes, and consequently, for repair.  Eventually, it will become a tablet.

Reply:
"subtable (plural subtables) A table of data corresponding to part of a greater parent table." ? :-) 

table_state can become compaction_group (makes sense in the compaction layer), but you're right compaction_group is not a good name for that class, living in replica namespace, storing a subset of a table.

Reply:
Compaction group could still be a good name if we think that compaction group refers to a LSM storage tree that is isolated and can be operated independently from others which belong to the same table. Then naturally compaction group interface will expose method for interacting with its memtable (which is in first level of the tree) and sstable set(s)  (which belong to higher levels).

Reply:
Maybe they can be already be called a tablet or simple_tablet. @avikivity what do you think?

Reply:
Or would it be appropriate only with tablet partitioning / replication mapping?

Reply:
Let's not hold our horses on naming.
I don't object to proceeding with `compaction_group`

Reply:
Let's go with compaction_group. We can always rename it later.